library/ssl_srv.c

```diff
@@ -3585,15 +3585,19 @@ static int ssl_parse_certificate_verify( mbedtls_ssl_context *ssl )
     /* Read the message without adding it to the checksum */
     do {
 
-        if( ( ret = mbedtls_ssl_read_record_layer( ssl ) ) != 0 )
+        do ret = mbedtls_ssl_read_record_layer( ssl );
+        while( ret == MBEDTLS_ERR_SSL_CONTINUE_PROCESSING );
```

Comment:
@RonEld Yes exactly, that's possible and a potential source for DoS - IOTSSL-1424 was raised to discuss and potentially rework the corresponding code paths. This PR keeps things as they are in this regard, only implementing the loop slightly differently.